modules/manifold/src/components/ui/segment-groups-control/segment-group-panel.js

```diff
@@ -0,0 +1,54 @@
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+
+import SegmentButtonGroup from './segment-button-group';
+import {SegmentPanel} from '../styled-components';
+import {updateSegmentGroups} from './utils';
+
+export default class SegmentGroupPanel extends PureComponent {
+  static propTypes = {
+    /** candidate segments shared across all segment groups: e.g., [0, 1, 2, 3] */
+    candidates: PropTypes.arrayOf(PropTypes.number),
+    /** selected segments for each segment group: e.g., [[1], [2, 3]] */
+    selected: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.number)),
+    /** callback function to surface the updated segment grouping */
+    onUpdateSegmentGroups: PropTypes.func,
+    /** colors indicatin each group */
+    colors: PropTypes.arrayOf(PropTypes.string),
+  };
+
+  static defaultProps = {
+    candidates: [],
+    selected: [[], []],
+    onUpdateSegmentGroups: () => {},
+    colors: ['#f00', '#000'],
+  };
+
+  _updateSelectedSegmentGroups = ({groupId, segmentId}) => {
+    const newSelected = updateSegmentGroups(
+      this.props.selected,
+      groupId,
+      segmentId
+    );
+    this.props.onUpdateSegmentGroups(newSelected);
+  };
+
+  render() {
+    const {colors, candidates, selected} = this.props;
+
+    return (
+      <SegmentPanel>
+        {[0, 1].map(groupId => (
```

Comment:
Is `[0, 1]` still valid here? From the code above it seems we can have more than two segments.
Is `[0, 1]` here meaning we can only have at most two for comparison?
Just wanted to double-check.

Reply:
Yes, `[0, 1]` is still valid because there can be many "segments", but only 2 "segment groups" for comparing against each other for now.